libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -32,20 +33,27 @@ const FloodSubCodec* = "/floodsub/1.0.0"
 type
   FloodSub* {.public.} = ref object of PubSub
     floodsub*: PeerTable      # topic to remote peer map
-    seen*: TimedCache[MessageId] # message id:s already seen on the network
-    seenSalt*: seq[byte]
+    seen*: TimedCache[SaltedId] # message id:s already seen on the network
+    seenSalt*: sha256
+      # The salt in this case is a partially updated SHA256 context pre-seeded
+      # with some random data
 
-proc hasSeen*(f: FloodSub, msgId: MessageId): bool =
-  f.seenSalt & msgId in f.seen
+proc salt*(f: FloodSub, msgId: MessageId): SaltedId =
+  var tmp = f.seenSalt
+  tmp.update(msgId)
+  SaltedId(data: tmp.finish())
 
-proc addSeen*(f: FloodSub, msgId: MessageId): bool =
+proc hasSeen*(f: FloodSub, msgId: SaltedId): bool =
```

Comment:
Should we update msgId to saltedMsgId? Same for the other places where the Type has changed but the corresponding param hasn't been renamed.

Reply:
513ee0bba5b7713745632a6609c90693baf8357e